framework/include/utils/MooseRandom.h

```diff
@@ -167,6 +167,11 @@ class MooseRandom
       it->second.first = it->second.second;
   }
 
+  LIBMESH_BEST_UNORDERED_MAP<unsigned int, std::pair<mt_state, mt_state> > getStates() { return _states; }
```

Comment:
Yes, it is set at libmesh configure time (check `libmesh/m4/unordered.m4` for the logic). There are five(!) different possible candidates.